src/v/coproc/pacemaker_offset_keeper.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+#include "coproc/ntp_context.h"
+#include "model/fundamental.h"
+#include "model/timeout_clock.h"
+#include "storage/log.h"
+#include "storage/snapshot.h"
+
+#include <seastar/core/gate.hh>
+#include <seastar/core/timer.hh>
+
+#include <chrono>
+namespace coproc {
+
+/// This class is responsible for flushing all of the coprocessors offsets to
+/// disk within a defined offset_persistence_interval.
+class pacemaker_offset_keeper {
+public:
+    /// Expects a reference to the world of tracked ntps, and a duration
+    /// Custom duration so far only used for unit tests
+    explicit pacemaker_offset_keeper(
+      storage::log_manager&,
+      ntp_context_cache&,
+      model::timeout_clock::duration d = std::chrono::minutes(5));
+
+    /// Starts the timer, when triggered offsets are collected, and serialized
+    /// to disk, this repeats until stop() is called
+    ss::future<> start();
```

Comment:
Not sure if i like the API might just make these free functions and have the simple fiber within pacemaker.h